lessons/beginners/str-methods/index.md

```diff
@@ -0,0 +1,109 @@
+# Řetězcové funkce a metody
+
+Řetězce umí všelijaké triky.
+Funkcí `len()` můžeš zjistit, jak je řetězec dlouhý;
+operátorem `in` pak jestli v sobě obsahuje daný podřetězec.
+
+<table class="table">
+    <tr>
+        <th>Zápis</th>
+        <th>Popis</th>
+        <th>Příklad</th>
+    </tr>
+    <tr>
+        <td><code>len(r)</code></td>
+        <td>Délka řetězce</td>
+        <td><code>len('čokoláda')</code></td>
+    </tr>
+    <tr>
+        <td><code>x&nbsp;in&nbsp;r</code></td>
+        <td>True pokud je řetězec <code>x</code> obsažen v <code>r</code></td>
+        <td><code>'oko' in 'čokoláda'</code></td>
+    </tr>
+    <tr>
+        <td><code>x&nbsp;not&nbsp;in&nbsp;r</code></td>
+        <td>Opak <code>x in r</code></td>
+        <td><code>'dub' not in 'čokoláda</code></td>
+    </tr>
+</table>
+
+Řetězce vždy berou v potaz velikost písmen,
+takže např. `'ČOKO' in 'čokoláda'` je `False`.
+Kdybys chtěl{{a}} porovnávat bez ohledu na velikost písmen,
+musel{{a}} bys oba řetězce převést třeba na malá písmena
+a pak je porovnat.
+
+A jak se převádí na malá písmena?
+K tomu budeme potřebovat další novou vlastnost Pythonu: metody.
+
+## Metody
+
+*Metoda* (angl. *method*) je jako funkce – něco, co se dá zavolat.
+Na rozdíl od funkce je svázaná s nějakým *objektem* (hodnotou).
+Volá se tak, že se za objekt napíše tečka,
+za ní jméno metody a za to celé se, jako u funkcí, připojí závorky
+s případnými argumenty.
+
+Řetězcové metody `upper()` a `lower()`
+převádí text na velká, respektive malá písmena.
+Zkus si to!
+
+```python
+retezec = 'Ahoj'
+print(retezec.upper())
+print(retezec.lower())
+print(retezec)
+```
+
+> [note]
+> Všimni si, že původní řetězec se nemění; metoda vrátí nový řetězec, ten
+> starý zůstává.
+>
+> To je obecná vlastnost řetězců v Pythonu: jednou existující řetězec se už
+> nedá změnit, dá se jen vytvořit nějaký odvozený.
```

Comment:
It was already said in the previous lesson, so I'd expect some continuity here to stress that those are two aspects of the same characteristic, for example: "Jistě si vzpomeneš, že ses už s touto vlastností setkal{{a}} při psaní funkce `zamen`."